frontend/packages/console-shared/src/components/health-checks/health-checks-alert.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Link } from 'react-router-dom';
+import { Alert, AlertActionCloseButton } from '@patternfly/react-core';
+import { ServiceModel as KnativeServiceModel } from '@console/knative-plugin';
+import { K8sResourceKind, K8sKind, referenceForModel } from '@console/internal/module/k8s';
+import {
+  DeploymentConfigModel,
+  DeploymentModel,
+  DaemonSetModel,
+  StatefulSetModel,
+} from '@console/internal/models';
+import './health-checks-alert.scss';
+import { STORAGE_PREFIX } from '../../constants';
+
+type HealthChecksAlertProps = {
+  resource: K8sResourceKind;
+  model: K8sKind;
+};
+
+const HIDE_HEALTH_CHECK_ALERT_FOR = `${STORAGE_PREFIX}/hide-health-check-alert-for`;
+
+const addHealthChecksRefs = [
+  referenceForModel(DeploymentConfigModel),
+  referenceForModel(DeploymentModel),
+  referenceForModel(DaemonSetModel),
+  referenceForModel(StatefulSetModel),
+  referenceForModel(KnativeServiceModel),
+];
+
+const HealthChecksAlert: React.FC<HealthChecksAlertProps> = ({ resource, model }) => {
+  const hiddenHealthCheckAlertFor = localStorage.getItem(HIDE_HEALTH_CHECK_ALERT_FOR);
```

Comment:
It would be better to have this in a `useEffect` with `[]` as dependency. That way it will only access `localStorage` on first render. Right now its doing it on every render which is not needed. In that effect you can simply do `setHideHealthCheckAlertFor(localStorage.getItem(HIDE_HEALTH_CHECK_ALERT_FOR))` to initialize your state with localStorage value. Everything else will remain same and work fine,

Reply:
Added ```useEffect``` and initialize the state into it.